### PR TITLE
Deprecate succeeding() and failing(), add failingThenComplete()

### LINKED ISSUE #77: The VertxTestContext.failing() method should release the latch
### Version

4.0.0-SNAPSHOT

### Context
I have created a test in order to ensure that my server is failing to startup due to a missing configuration.
So the MainVerticle is being failed with an exception. Then the MainVerticleIT is supposed to pass ok since it is waiting for a failure, but the test is wrongly failing with a timeout instead.

### Do you have a reproducer?

https://github.com/cvgaviao/vertx-junit5-investigation

### Steps to reproduce

1. just run the included test

### Extra
I was able to pass the test using the failing method with parameter...
```
vertx.deployVerticle(new MainVerticle(), testContext.failing((o) -> {
            testContext.completeNow();
        }));
```

### PR DESCRIPTION
**Pull request by julianladisch**
Using the parameter-less method VertxTestContext#succeeding()
is almost always unintended and should be replaced by

* completing()
* succeeding(value -> checkpoint.flag())
* succeeding(value -> { more testing code })
* succeeding(value -> {})

The same for failing().

Deprecating for removal the methods succeeding() and failing() improves
VertxTestContext's ergonomics because people are no longer
tempted to use them. The replacement either fixes an unintended
bug in people's unit test code or is more readable because
it explicitly states the intention.

Example for unintended succeeding() usage that is not executed:
https://github.com/reactiverse/reactiverse-junit5-extensions/pull/3/files

The new method failingThenComplete() is a short-cut for
failing(exception -> vertxTestContext.completeNow())
and parallels completing() that is an existing short-cut for
succeeding(value -> vertxTestContext.completeNow()).

Note that vertx-unit's TestContext#asyncAssertSuccess() and
TestContext#asyncAssertFailure() are different because they
complete the test context by design. People switching from
vertx-unit to vertx-junit5 will not fall into this trap if we
remove succeeding() and failing().

Fixes #77.

### CODE REVIEW

**Comment by jponge** [comment]
Hi @julianladisch, this makes sense indeed.

However, we cannot simply remove methods like this. We should mark them as deprecated, then keep them until at least Vert.x 4.1 / 4.2.

/cc @vietj @slinkydeveloper

**Comment by julianladisch** [comment]
I've deprecated succeeding() and failing() instead of removing them.

File: src/main/java/io/vertx/junit5/VertxTestContext.java [hunk 1]
```diff
@@ -238,6 +246,7 @@ public synchronized Checkpoint checkpoint(int requiredNumberOfPasses) {
    *
    * @param <T> the asynchronous result type.
    * @return the handler.
+   * @see #failingThenComplete()
    */
   public <T> Handler<AsyncResult<T>> completing() {
```
Comment (slinkydeveloper) [inline src/main/java/io/vertx/junit5/VertxTestContext.java:251]: Uhm i get your point, how you feel about naming this as `succedingThenComplete` to be consistent with `failingThenComplete`? Also @jponge

Comment (jponge) [reply src/main/java/io/vertx/junit5/VertxTestContext.java:251]: Yes `succedingThenComplete` would make sense alongside a deprecated `succeeding`

**Comment by jponge** [comment]
Ok, I'll merge

**Comment by jponge** [comment]
Thanks @julianladisch

**Pull request merged**